src/main/java/org/seqdoop/hadoop_bam/VCFInputFormat.java

```diff
@@ -149,6 +155,33 @@ public VCFFormat getFormat(final Path path) {
 		return fmt;
 	}
 
+	@Override
+	protected boolean isSplitable(JobContext context, Path filename) {
+		Configuration conf = context.getConfiguration();
+		final CompressionCodec codec =
+				new CompressionCodecFactory(context.getConfiguration()).getCodec(filename);
+		if (codec == null) {
+			return true;
+		}
+		if (codec instanceof BGZFCodec) {
+			boolean splittable;
+			try {
+				try (FSDataInputStream in = filename.getFileSystem(conf).open(filename)) {
+					splittable = VCFFormat.isBGZF(new BufferedInputStream(in));
+				}
+			} catch (IOException e) {
+				// can't determine if BGZF or GZIP, conservatively assume latter
+				splittable = false;
+			}
+			if (!splittable) {
+				System.err.printf("Warning: %s is not splittable, consider using block " +
+						"compressed gzip (BGZF).\n", filename);
+			}
+			return splittable;
+		}
+		return codec instanceof SplittableCompressionCodec;
+	}
+
```

Comment:
Yeah, thats what I meant. It would just be a shame to fall back to gzip since that won't split right ? Certainly not a requirement, just a nice to have.